lib/workers/repository/process/container-vulnerabilities.ts

```diff
@@ -0,0 +1,478 @@
+import { OsvOffline } from '@renovatebot/osv-offline';
+import type { Osv } from '@renovatebot/osv-offline';
+import is from '@sindresorhus/is';
+import type { CvssScore } from 'vuln-vects';
+import { parseCvssVector } from 'vuln-vects';
+import { getManagerConfig, mergeChildConfig } from '../../../config';
+import type { PackageRule, RenovateConfig } from '../../../config/types';
+import { logger } from '../../../logger';
+import { DockerDatasource } from '../../../modules/datasource/docker';
+import type {
+  PackageDependency,
+  PackageFile,
+} from '../../../modules/manager/types';
+import { findGithubToken } from '../../../util/check-token';
+import { find } from '../../../util/host-rules';
+import { sanitizeMarkdown } from '../../../util/markdown';
+import * as p from '../../../util/promises';
+import { regEx } from '../../../util/regex';
+import { titleCase } from '../../../util/string';
+import type { ContainerVulnerability, SeverityDetails } from './types';
+
+export class ContainerVulnerabilities {
+  /* tslint:disable:no-unused-variable */
+  private osvOffline: OsvOffline | undefined;
+  private dockerDatasource: DockerDatasource;
+
+  private constructor() {
+    this.dockerDatasource = new DockerDatasource();
+  }
+
+  private async initialize(): Promise<void> {
+    // hard-coded logic to use authentication for github.com based on the githubToken for api.github.com
+    const token = findGithubToken(
+      find({
+        hostType: 'github',
+        url: 'https://api.github.com/',
+      }),
+    );
+
+    this.osvOffline = await OsvOffline.create(token);
+  }
+
+  static async create(): Promise<ContainerVulnerabilities> {
+    const instance = new ContainerVulnerabilities();
+    await instance.initialize();
+    return instance;
+  }
+
+  async appendVulnerabilityPackageRules(
+    config: RenovateConfig,
+    packageFiles: Record<string, PackageFile[]>,
+  ): Promise<void> {
+    const dependencyVulnerabilities = await this.fetchDependencyVulnerabilities(
+      config,
+      packageFiles,
+    );
+
+    config.packageRules ??= [];
+
+    for (const vulnerability of dependencyVulnerabilities) {
+      const rule = this.vulnerabilityToPackageRules(vulnerability);
+      if (is.nullOrUndefined(rule)) {
+        continue;
+      }
+      config.packageRules.push(rule);
+    }
+  }
+
+  private async fetchDependencyVulnerabilities(
+    config: RenovateConfig,
+    packageFiles: Record<string, PackageFile[]>,
+  ): Promise<ContainerVulnerability[]> {
+    const managers = Object.keys(packageFiles);
+
+    // TODO: should we also include other docker managers: devcontainer, docker-compose ?
```

Comment:
We don't yet support those in MintMaker but we support others like tekton and even a custom regex manager can work with docker datasource/versioning.